src/frontend/classic/utils/storage.ts

```diff
@@ -0,0 +1,46 @@
+export type SettingValue = string | number | boolean;
+export type SettingExtras = ((extrasArg?: SettingValue) => void) | undefined
+export type SettingValidator = ((value: SettingValue) => boolean) | undefined
+
+export const onSettingChange = (
+    key: string,
+    value: SettingValue,
+    validator: SettingValidator,
+    extras: SettingExtras,
+    extrasArg: SettingValue | undefined
+): boolean => {
+    if (!validator || (validator && validator(value))) {
+        localStorage.setItem(key, value.toString());
```

Comment:
The `localStorage` is assumed not to be persistent and may (theoretically) be cleared whenever chrome feels like it. HakuNeko engine will provide certain levels of storages in the future...

Reply:
I've review and it should be shifted to svelte/store that will be linked to hakuneko storage (but will be linked to local storage while we're moking).